### DISCUSSION TITLE
How to plot top species in Triplot RDA?

### DISCUSSION THREAD

**Discussion by Stabilo12** [vegandevs/vegan#474] [Q&A]
Hi, 

I'm quite new to R and hope somebody can help me. I would like to plot only the top e.g. five species in an RDA figure, as the other species are all overlapping and make the figure messy (see example). 

I tried to use: select = abs(sum_rda$sp) > 0.5
However, it doesn't work but also does not show any error message. 

```r
data(varespec)
data(varechem)
ord <- rda(varespec)
summary(ord)
sum_rda <- summary(ord)

plot(ord, scaling=2, main="Triplot RDA", type="none", xlab=c("RDA1"), ylab=c("RDA2"), 
     xlim=c(-10,20), ylim=c(-10,10))
points(scores(ord, display="sites", choices=c(1,2), scaling=2), 
       pch=21, col="black", cex=1.2)
arrows(0,0, 
       scores(ord, select = abs(sum_rda$sp) > 0.5, display="species", choices=c(1),
              scaling=2)*2,  
       scores(ord, select = abs(sum_rda$sp) > 0.5, display="species", choices=c(2),
              scaling=2)*2,
       col="black",length=0, lty = 1)

text(scores(ord, select = abs(sum_rda$sp) > 0.5,  display="species", choices=c(1),
            scaling=2)*2.1,
     scores(ord, select = abs(sum_rda$sp) > 0.5,  display="species", choices=c(2), 
            scaling=2)*2.1,
     labels=rownames(scores(ord, display="species", scaling=2)),
     col="black", cex=0.8)   
```

![image](https://user-images.githubusercontent.com/37101233/72667466-6f5bbd80-3a1c-11ea-826b-aab32ddc3328.png)

Thanks for considering my request

**Comment by Imnotdogno** [comment]
Hi did you resolve this?
Can you please tell me how to plot only the most important species?

**Comment by jarioksa** [comment]
Seems to be a long and stale issue. The reason that the `select` did not work was that it was wrongly used. You are not using **vegan** functions `text` and `points`, but default **R** functions. You are just extracting the scores with **vegan**, but the `scores` function has no `select` argument: it will return all asked scores. If you want to take a subset of those in standard **R** functions, you need to use standard **R** tools, such as indexing, as they know no `select`.

I would like to suggest a different selections criterion: now you use axiswise length limit, meaning that the arrows should be within a square. I think it is better to base this test on arrow length, that is, to have arrows within a circle. Get the arrow lengths in 2-dim with
```r
splen <- sqrt(rowSums(sum_rda$sp[,1:2]^2)) # your sum_rda object
```
Now to the solution: **vegan** `plot` functions return an invisible `ordiplot` object which you can catch, and this object has generic `text` and `points`  functions that also have `select` and `arrows` argument. So this is a complete sequence:
```r
library(vegan)
## copied from your message
data(varespec, varechem)
ord <- rda(varespec)
## modified from this on: catch the plot structure in pl
pl <- plot(ord, scaling=2, main="Triplot RDA", type="none", xlab=c("RDA1"), ylab=c("RDA2"), 
     xlim=c(-10,20), ylim=c(-10,10))
splen <- sqrt(rowSums(pl$species^2))  # species distance from the origin in pl[ot]
points(pl, "sites", pch=21, col="black", cex=1.2)
text(pl, "species", select = splen > 0.5, arrow=TRUE, length=0.05) # harmless warning on length argument, but works
## Alternatively you can use magrittr pipes (if you have splen  or select vector calculated by other means):
library(magrittr)
plot(ord, scaling=2, main="Triplot RDA", type="none", xlab=c("RDA1"), ylab=c("RDA2"), 
   xlim=c(-10,20), ylim=c(-10,10)) %>%
 points(what="sites", pch=21, col="black", cex=1.2) %>%
 text(what="species",  select = splen > 0.5, arrow=TRUE, length=0.05) # you need splen for this to work
```
